src/watchdog.rs

```diff
@@ -0,0 +1,31 @@
+//! Traits for interactions with a processors watchdog timer.
+
+
+
+/// Kicks an existing watchdog to ensure the processor isn't reset.
+#[cfg(feature = "unproven")]
+pub trait Watchdog {
+    /// Triggers the watchdog. This must be done once the watchdog is started
+    /// to prevent the processor being reset.
+    fn kick(&mut self);
```

Comment:
This is just a nitpick, and maybe I'm squeamish, but I don't like the thought of kicking a dog :)

In the LPC82x documentation, this operation is called "feed"; nRF52 calls this "reload". I think "reload" is a better and more descriptive name.